lib/LaTeXML/Core/State.pm

```diff
@@ -507,7 +507,7 @@ sub pushFrame {
 sub popFrame {
   my ($self) = @_;
   if ($$self{undo}[0]{_FRAME_LOCK_}) {
-    Fatal('unexpected', '<endgroup>', $self->getStomach,
+    Error('unexpected', '<endgroup>', $self->getStomach,
```

Comment:
Did you try triggering this one? This is one place where you can hit an infinite loop (or just 100 errors really) since `\end{document}` in LaTeX.pool loops over popFrame.